core/heap.c

```diff
@@ -1401,6 +1401,9 @@ vmm_heap_init()
     if (DYNAMO_OPTION(vm_reserve)) {
         vmm_heap_unit_init(&heapmgt->vmheap, DYNAMO_OPTION(vm_size));
     }
+    dynamo_vm_areas_lock();
+    heap_exiting = false;
```

Comment:
Reading a bit closer, you meant the latter. Implemented similarly to #2157.
Also, it's moot grabbing dynamo_vm_areas_lock() at that point, because it's already NULL and this is the only DR thread left.